osu.Game/Screens/Edit/Timing/TapTimingControl.cs

```diff
@@ -224,6 +230,11 @@ private void adjustBpm(double adjust)
                 return;
 
             timing.BeatLength = 60000 / (timing.BPM + adjust);
+
+            if (beatmap.AdjustNotesOnOffsetBPMChange.Value)
+                timing.SetHitObjectBPM(beatmap, 60000 / (timing.BPM - adjust));
```

Comment:
This kind of subtraction is asking for trouble because floating point numbers don't behave like real world math does and you can introduce subtle rounding errors this way. It can be the case that `timing.BPM + adjust - adjust != timing.BPM`, and this is even worse because there are like three divisions by 60000 along the way.

Just store the old value and pass it in.